### PR TITLE
mavlink mode review

### PR DESCRIPTION
**Pull request by dagar**
The existing mavlink modes continue to grow without real consideration for the particular usage.

For example in `normal` (the typical default) we now have enormous messages like `ODOMETRY`, `TRAJECTORY_REPRESENTATION_WAYPOINTS`, etc that on typical inexpensive serial radios end up significantly throttling the connection to the point there's very little of the other telemetry messages.

We need to do a pass and decide what each of these modes is actually for and then optimize that case.

Related - https://github.com/PX4/Firmware/issues/12974

Messages to review in detail
 - [ ] HIGHRES_IMU
 - [ ] OBSTACLE_DISTANCE
 - [ ] ODOMETRY
 - [ ] TRAJECTORY_REPRESENTATION_WAYPOINTS

@TSC21 @jkflying @mrivi @baumanta can you comment on the need for any of these messages in the mavlink `normal` mode?

### CODE REVIEW

**Comment by dagar** [comment]
Additionally this is why mavlink cpu usage continues to grow and is now a significant portion of any setup with a companion computer. Serial TX DMA will help somewhat, but it's not a magic solution. A large portion of the time is spent accumulating data, calculation the CRC, etc.

File: src/modules/mavlink/mavlink_main.cpp [hunk 1]
```diff
@@ -1620,26 +1621,21 @@ Mavlink::configure_streams_to_default(const char *configure_single_stream)
 		configure_stream_local("GLOBAL_POSITION_INT", 5.0f);
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_main.cpp]: Can we lower this one?

File: src/modules/mavlink/mavlink_main.cpp [hunk 2]
```diff
@@ -1607,8 +1607,9 @@ Mavlink::configure_streams_to_default(const char *configure_single_stream)
 	case MAVLINK_MODE_NORMAL:
 		configure_stream_local("ADSB_VEHICLE", unlimited_rate);
 		configure_stream_local("ALTITUDE", 1.0f);
-		configure_stream_local("ATTITUDE", 20.0f);
+		configure_stream_local("ATTITUDE", 15.0f);
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/modules/mavlink/mavlink_main.cpp]: The attitude indicator is updated from ATTITUDE. This is the main part that "feels" responsive or not when moving a vehicle around.

File: src/modules/mavlink/mavlink_main.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/mavlink/mavlink_main.cpp]: I'd say yes if we can strip out the others. The messages are scaled back relative to the initial value, so having this slightly higher than the others is somewhat of a coping mechanism to get the data through at 1 or 2 Hz.

File: src/modules/mavlink/mavlink_main.cpp [hunk 3]
```diff
@@ -1607,8 +1607,9 @@ Mavlink::configure_streams_to_default(const char *configure_single_stream)
 	case MAVLINK_MODE_NORMAL:
 		configure_stream_local("ADSB_VEHICLE", unlimited_rate);
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/modules/mavlink/mavlink_main.cpp]: Probably not worth worrying about unless you're aware of someone using this in production that can comment.

File: src/modules/mavlink/mavlink_main.cpp [hunk 4]
```diff
@@ -1620,26 +1621,21 @@ Mavlink::configure_streams_to_default(const char *configure_single_stream)
...
+		configure_stream_local("OBSTACLE_DISTANCE", 1.0f);
+		configure_stream_local("ORBIT_EXECUTION_STATUS", 2.0f);
 		configure_stream_local("PING", 0.1f);
-		configure_stream_local("POSITION_TARGET_GLOBAL_INT", 1.5f);
+		configure_stream_local("POSITION_TARGET_GLOBAL_INT", 1.0f);
 		configure_stream_local("POSITION_TARGET_LOCAL_NED", 1.5f);
 		configure_stream_local("RC_CHANNELS", 5.0f);
 		configure_stream_local("SERVO_OUTPUT_RAW_0", 1.0f);
 		configure_stream_local("SYS_STATUS", 1.0f);
-		configure_stream_local("TRAJECTORY_REPRESENTATION_WAYPOINTS", 5.0f);
```
Comment (jkflying) [inline src/modules/mavlink/mavlink_main.cpp]: @mrivi do we need to add this to something else instead?

File: src/modules/mavlink/mavlink_main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (julianoes) [reply src/modules/mavlink/mavlink_main.cpp]: Oh, you're right!

File: src/modules/mavlink/mavlink_main.cpp [hunk 5]
```diff
@@ -1620,26 +1621,21 @@ Mavlink::configure_streams_to_default(const char *configure_single_stream)
 		configure_stream_local("GLOBAL_POSITION_INT", 5.0f);
 		configure_stream_local("GPS2_RAW", 1.0f);
 		configure_stream_local("GPS_RAW_INT", 1.0f);
-		configure_stream_local("HIGHRES_IMU", 1.5f);
 		configure_stream_local("HOME_POSITION", 0.5f);
 		configure_stream_local("LOCAL_POSITION_NED", 1.0f);
 		configure_stream_local("NAMED_VALUE_FLOAT", 1.0f);
-		configure_stream_local("NAV_CONTROLLER_OUTPUT", 1.5f);
-		configure_stream_local("OBSTACLE_DISTANCE", 5.0f);
-		configure_stream_local("ODOMETRY", 3.0f);
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/modules/mavlink/mavlink_main.cpp]: Exactly...

**Comment by dagar** [comment]
I'm going to get this into master and we'll keep going. If anyone has a problem with the mavlink mode selection please contact me directly and we'll work through adding additional modes or more granularity.

**Pull request merged**